[quant][pt2e] Add support for FixedQParamsQuantizationSpec

jerryzh168 opened the pull request:
Summary:
This PR adds support for FixedQParamsQuantizationSpec:

```
dataclass(eq=True, frozen=True)
class FixedQParamsQuantizationSpec(QuantizationSpecBase):
    dtype: torch.dtype
    scale: float
    zero_point: int
    quant_min: Optional[int] = None
    quant_max: Optional[int] = None
    qscheme: Optional[torch.qscheme] = None
```

This is useful to define quantization spec for operators like sigmoid which has predefined and fixed scale/zero_point

Test Plan:
```
buck2 test mode/opt caffe2/test:quantization_pt2e -- 'caffe2/test:quantization_pt2e'
buck2 test mode/opt caffe2/test:quantization_pt2e -- --exact 'caffe2/test:quantization_pt2e - test_fixed_qparams_qspec (quantization.pt2e.test_quantize_pt2e.TestQuantizePT2E)'
```

Reviewed By: kimishpatel

Differential Revision: D46153082

leslie-fang-intel commented on test/quantization/pt2e/test_quantize_pt2e.py:
```diff
@@ -365,6 +366,76 @@ def get_supported_operators(cls) -> List[OperatorConfig]:
+                            quant_min=0,
+                            quant_max=255,
+                            qscheme=torch.per_tensor_affine,
+                            scale=2.0 / 256.0,
```
Hi @jerryzh168, May I ask a question about this fixed scale and zp value of `sigmoid`? The output range of `sigmoid` is among 0-1, so why we set the fixed scale as `2.0 / 256.0` and fixed zp as `128`.

jerryzh168 replied:
oh sorry I think this is not correct, I was trying to copy from https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/backend_config/_common_operator_config_utils.py but was writing down the wrong parameters, thanks for the comment!

jerryzh168 committed:
[quant][pt2e] Add support for FixedQParamsQuantizationSpec (#102439)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/102439

This PR adds support for FixedQParamsQuantizationSpec:

```
dataclass(eq=True, frozen=True)
class FixedQParamsQuantizationSpec(QuantizationSpecBase):
    dtype: torch.dtype
    scale: float
    zero_point: int
    quant_min: Optional[int] = None
    quant_max: Optional[int] = None
    qscheme: Optional[torch.qscheme] = None
```

This is useful to define quantization spec for operators like sigmoid which has predefined and fixed scale/zero_point

Test Plan:
```
buck2 test mode/opt caffe2/test:quantization_pt2e -- 'caffe2/test:quantization_pt2e'
buck2 test mode/opt caffe2/test:quantization_pt2e -- --exact 'caffe2/test:quantization_pt2e - test_fixed_qparams_qspec (quantization.pt2e.test_quantize_pt2e.TestQuantizePT2E)'
```

Reviewed By: kimishpatel

Differential Revision: D46153082

fbshipit-source-id: efe352792d9b178648423da04264b89f7bf35923
